### ISSUE TITLE
Feature: HOTP Auth

### ISSUE THREAD

**Issue by Properrr** [mooltipass/minible#235]
### Missing feature
HOTP Auth

Would be cool to have HOTP

**Comment by limpkin** [comment]
Hello!
Just to make sure I follow you: you'd like to see OTP and **not** TOTP on the device?

**Comment by Properrr** [comment]
Hello @limpkin 
HOTP or TOTP would be great as well

**Comment by limpkin** [comment]
but the mini BLE does handle TOTP :)

**Comment by limpkin** [comment]
so should we rename this issue for HOTP?
however, if using HOTP.... syncing multiple devices together will create issues :/

**Comment by 0x0ptr** [comment]
@Properrr How widespread is the use of HOTP? I have not seen a site yet that uses HOTP but that does not mean it doesn't exists of course. Can you give me example of web sites you have encountered that support only HOTP (and not TOTP)? Any test sites that support HOTP?

Like @limpkin said this will break when you have multiple devices. The counter would need to be somehow carefully synch'ed between the devices....Also, when do you increment the counter? After MiniBLE did the autotyping? After displaying? Prompt to update the counter?

**Comment by 0x0ptr** [comment]
@Properrr Planning to close this issue within a week due to inactivity/no response.

**Comment by Arffeh** [comment]
Funnily enough, I was eyeballing HOTP about 3 nights ago, and found the
ancient github issue before the MP-BLE existed.

I was looking at it due to some IdP products using it as an option.
(Keycloak comes to mind)

**Comment by nlutsenko** [comment]
Coming from almost a decade usage of products by Yubico...
I can't be more than happy with MooltiPass, at the same time, the primary use case that I have is work-related, and for better or for worse, can't use U2F for it. Meaning that HOTP, the old and rusty, is what I use most of the time for 2-factor, and super surprised for it not being available.


One more data point:

Duo, as in https://duo.com.
These folks are multi-key 2fac provider, and they support most of Yubikeys + any OATH HOTP keys, but seems like not built-in others, resulting in pretty much dicouraging use of MooltiPass for it. (more here: https://duo.com/product/multi-factor-authentication-mfa/authentication-methods/tokens-and-passcodes)

---

Basically, literally can't wait to test out HOTP, how can we help to move this along?

**Comment by 0x0ptr** [comment]
Just to confirm that we are talking about the same thing. Your use case is mostly for logging in to work account and they only support HOTP - COUNTER based One-Time-Password (OTP) and NOT TOTP (Time-based One-time Password Algorithm)? And they are willing to provide you with the HOTP secret to use on any device [not managed by your IT department]? They don't just give you an "RSA SecurID" or a Yubikey (for U2F/FIDO) (for example) for the purpose of authenticating?

As discussed above there are multiple issues related to implementing this on Mooltipass MiniBLE as outlined above:
1) This will break when you have multiple devices. The counter would need to be somehow carefully synch'ed between the devices
2) When do you increment the counter? After MiniBLE did the auto-typing? After displaying the HTOP? Prompt to update the counter?

**Comment by limpkin** [comment]
closing due to inactivity.
